INSTALL

```diff
@@ -565,6 +592,10 @@
                    CPPINCLUDES     List of CPP inclusion directories, separated
                                    the same way as for CPPDEFINES.  This can
                                    be used in place of -I.
+                   DEVRANDOM       A comma separated list of strings that
+                                   specify pathes of random devices which are
+                                   searched in the given order for seeding the
+                                   random generator.
```

Comment:
Wait, what?  I can see no such thing...

Reply:
I added documentation about `DEVRANDOM`, because I noticed that documentation of `--with-rand-seed=devrandom` was inaccurate: It said
```
    devrandom:   Use the /dev/urandom device.
```
This is false, because the random device is chosen from a comma separated path list, which is `#define`d by the preprocessor constant `DEVRANDOM` in [e_os, line 29 ff](https://github.com/openssl/openssl/blob/master/e_os.h#L29). The `#define` is guarded by an `#ifndef`, so (in theory) it can be overwritten on the command line. My understanding was that it could be added to the configure arguments as `-DDEVRANDOM=...`. If that is wrong or too much level of detail, I'll remove it again.

Think over it twice, it is rather clumsy to have to modify both `--with-rand-seed` and `DEVRANDOM`. In particular, because of the necessity to quote the double quotes when passing `DEVRANDOM` on the commandline is rather messy.  (BTW, the same holds for `--with-rand=egd` and `DEVRANDOM_EGD`).

In fact, it would be much better to make `--with-rand-seed` more flexible, so one could configure the random sources directly using `--with-rand-seed` and not having to fiddle around with `DEVRANDOM`. A possible solution would be to provide the possibility for adding an optional path list as arguments to `devrandom` and `egd`. Here I chose `+` as path separator, because `,` already separates the seed methods. (The choice is arbitrary, maybe you have a better choice?)

```
--with-rand-seed=seed1[,seed2,....]

	devrandom[=/dev/devrandom+/dev/urandom+...]
	egd[=/dev/devrandom+/dev/urandom+...]	
```

OpenSSL would proved proper defaults, when `--with-rand-seed=os` is used. For example, it would imply

```
   --with-rand-seed="devrandom:/dev/urandom+/dev/random,/dev/srandom" on S390, and
   --with-rand-seed="devrandom:/dev/prandom+/dev/urandom+/dev/hwrng+/dev+random" otherwise.
```

But for such a change I would need your help (and/or help by @paulidale), because I'm not firm with perl and the build internals. Such a change comes practically too late for 1.1.1, so it would probably be better to leave it 1.2. (or not do it at all...)

Please let me know what your preferred solution for the moment ist, whether I should revert my mentioning of `DEVRANDOM` or how I should change its documentation. I will be happy to follow your advice.

Reply:
Note: had to make some minor corrections to my previous post. It's better to read it online.

Reply:
But this part of `INSTALL` isn't about cpp macros, it's about "make variables" that can be passed to Configure.  In other words, what you're saying is that there would be a variable that you could control like this:

```
./config DEVRANDOM='/dev/random /dev/srandom'
```

There is no support for that variable.

Reply:
Ok, I'll remove that hunk again. But now I have to leave, I'll do it later...

Reply:
Or, alternatively:  Would it be a great effort for you to add such a make variable, which would create the `-DDEVRANDOM=...` compiler argument automatically? (And same for `DEVRANDOM_EGD`?)

Reply:
> Or, alternatively:  Would it be a great effort for you to add such a make variable, which would create the `-DDEVRANDOM=...` compiler argument automatically? (And same for `DEVRANDOM_EGD`?)

The diverse variables that are used to set CPP flags is already a complex enough matter, I see no reason why we should add to it, especially if the user can already do this:

```
./config -DDEVRANDOM='"/dev/random","/dev/srandom"'
```

or

```
./config CPPDEFINES='DEVRANDOM="/dev/random","/dev/srandom"'
```

In that case, I *much* prefer your idea of making the `--with-seed-source` argument more flexible. Trouble is that we're going into beta on tuesday, time's pressing...

Reply:
Commit 6f82183662e8 reverts DEVRANDOM entry in make variable section.